### PR TITLE
Fix browser popup warning when deleting image

### PR DESCRIPTION
**Pull request by sam365724**
Small fix for avoiding an additional popup when clicking on image delete.
Also handles error case when deleting image fails.
Adds the user supplied image filename for clear identification (this will be relevant for multiple image upload coming).

image

### CODE REVIEW

**Comment by sam365724** [comment]
@dessalines How do i create entries for newly created language strings?

**Comment by sam365724** [comment]
> @dessalines How do i create entries for newly created language strings? Can you please add: ` "failed_to_delete_picture": "Failed to delete picture.",`

https://github.com/LemmyNet/lemmy-translations/pull/29

File: src/shared/components/common/markdown-textarea.tsx [hunk 1]
```diff
@@ -386,8 +386,9 @@ export class MarkdownTextArea extends Component<
           let textarea: any = document.getElementById(i.id);
           autosize.update(textarea);
           pictrsDeleteToast(
-            i18n.t("click_to_delete_picture"),
-            i18n.t("picture_deleted"),
+            i18n.t("click_to_delete_picture").concat(' (', file.name,')'),
```
Comment (dessalines) [inline src/shared/components/common/markdown-textarea.tsx:389]: Use string literals, not concat. I'd prefer these to be colons as well, so:

```
`${i18n.t("click_to_delete_picture)}: ${file.name}`
```

Same for the other two.

File: src/shared/utils.ts [hunk 2]
```diff
@@ -552,9 +553,16 @@ export function pictrsDeleteToast(
       duration: 10000,
       onClick: () => {
         if (toast) {
-          window.location.replace(deleteUrl);
-          alert(deletePictureText);
-          toast.hideToast();
+          fetch(deleteUrl, {})
```
Comment (dessalines) [inline src/shared/utils.ts:556]: Why are you doing a fetch here, doesn't it delete the picture elsewhere? This should just be the toast.

File: src/shared/components/common/markdown-textarea.tsx (hunk 1 above)
Comment (sam365724) [reply src/shared/components/common/markdown-textarea.tsx:389]: Alright, we need to update translations then. I'll do a pull request.

Comment (sam365724) [reply src/shared/components/common/markdown-textarea.tsx:389]: I did this change in the asyn image upload branch now though.

File: src/shared/utils.ts (hunk 2 above)
Comment (sam365724) [reply src/shared/utils.ts:556]: No, it doesn't here we had a simple window.location() which the browser does not like.

Comment (dessalines) [reply src/shared/utils.ts:556]: In that case use a callback function for onClick, because this should really just have toast-related code.

Comment (sam365724) [reply src/shared/utils.ts:556]: I don't really understand the difference, can you make the change or a proposal?

Comment (dessalines) [reply src/shared/utils.ts:556]:
```ts
export function pictrsDeleteToast(...
 onClick: () => void
) {
...
  onClick: () => {
    toast.hideToast();
    onClick();
  }
}
...
pictrsDeleteToast(.... () => { do fetching stuff here }
```

Comment (sam365724) [reply src/shared/utils.ts:556]: I think you misunderstand the logic here. The picture is deleted when the user clicks on the toast message itself. So it must be inside onClick?! Please just fix it yourself, I'm losing interest like this. It's not worth it for such small but annoying bug. :(

**Pull request merged**